tests/v2_validation/cattlevalidationtest/core/test_cattle_upgrade.py

```diff
@@ -0,0 +1,291 @@
+    client_port = port + "0"
+    domain = dom_list[0]
+    test_ssl_client_con = create_client_container_for_ssh(client, client_port)
+    print "***TEST CLIENT CONTAINER***"
+    print test_ssl_client_con["port"]
+    print test_ssl_client_con["host"]
+    print test_ssl_client_con["container"]
+    validate_lb_services_ssl(admin_client, client, test_ssl_client_con,
+                             stack, [service1, service2], ssllbservice,
+                             port, ssl_port,
+                             domain)
+
+    # Validate DNS Service
+    stack, servicewithexposedports = get_env_service_by_name(
+                                     admin_client, stack1,
+                                     "servicewithexposedports")
+    assert servicewithexposedports['state'] == "active"
+    assert servicewithexposedports.scale == 1
+    exposedport = "400" + str(portsuffixnum)
+    validate_dns_service(
+        admin_client, servicewithexposedports, [service1],
+        exposedport, "myalias")
+
+    # Validate External Service
+    stack, extservicetohostname = get_env_service_by_name(
+                                    admin_client,
+                                    stack1,
+                                    "extservicetohostname")
+    validate_external_service_for_hostname(
+        admin_client, servicewithexposedports,
+        [extservicetohostname], exposedport)
+
+    # Validate Service with Link
+    stack, servicewithlink = get_env_service_by_name(admin_client,
+                                                     stack1, "servicewithlink")
+    assert servicewithlink['state'] == "active"
+    assert servicewithlink.scale == 1
+    servicelinkexposedport = "500" + str(portsuffixnum)
+    validate_linked_service(admin_client, servicewithlink,
+                            [service2], servicelinkexposedport,
+                            linkName="mylink")
+
+    # Validate LB Service in NewStack
+    stack, newstacklbservice = get_env_service_by_name(client, stack2,
+                                                       "newstacklb")
+    assert newstacklbservice['state'] == "active"
+    assert newstacklbservice.scale == 1
+
+    validate_lb_service(admin_client, client, newstacklbservice,
+                        "9001", [service2])
+
+    # Validate Service with Link in NewStack [The link is
+    # pointing to a service in Default Stack]
+    stack, newstackservice1 = get_env_service_by_name(client, stack2,
+                                                      "newstackservice1")
+    assert newstackservice1['state'] == "active"
+    assert newstackservice1.scale == 1
+
+    stack, newstackservicewithlink = get_env_service_by_name(
+            client, stack2, "newstackservicewithlink")
+    assert newstackservicewithlink['state'] == "active"
+    assert newstackservicewithlink.scale == 1
+
+    validate_linked_service(admin_client, newstackservicewithlink,
+                            [service1], "9002",
```

Comment:
Add validation checks to ensure services can be accessed by their name from containers of other services: dig servicename dig servicename.stackname

Test this for within stack as well as cross stack